### PR TITLE
auto count and simple_skins support

### PR DESCRIPTION
**Pull request by ghost**
Loop through textures file to count them = super easy to add new textures, its just a matter of placing file in the right folder with the right name
Support for simple_skins mod = more textures and ability to change in game and prevent duplicates when adding new textures

This loop/hash thing made me think about making a texture related rpg style character description, perhaps stored in a meta text file like the ones in simple_skins ... or maybe that would just be boring

### CODE REVIEW

File: init.lua [hunk 1]
```diff
@@ -3,17 +3,58 @@ minetest.register_on_joinplayer(function(player)
...
   
+	h = {}
+  -- prefix of textures files
+	h.sprefix = "player_hash_skin_"
+  -- mod providing textures files
+	h.smodpath = minetest.get_modpath("player_hash_skin")
+  -- start count of texture files
+	h.s = 0
+  -- number of texture file
+	h.n = 8
```
Comment (Hiradur) [inline init.lua]: What is this for? This value of `h.n` is not used anywhere.

File: init.lua [hunk 2]
```diff
@@ -3,17 +3,58 @@ minetest.register_on_joinplayer(function(player)
...
+	h.n = h.n - h.s
+	
+ 	minetest.log("action", "[player_hash_skin] "..h.n.." skins found")
+
+	-- hash
+ 	hash = hash%h.n
+ 	minetest.log("action", "[player_hash_skin] hash =  "..hash..".")
+
+
+	-- ouvre le fichier
```
Comment (Hiradur) [inline init.lua]: Can you translate this?

**Review by Hiradur — CHANGES_REQUESTED** [review]

**Comment by Hiradur** [comment]
Thanks for the PR, I like the changes. There are just two minor issues with it (see review).

> This loop/hash thing made me think about making a texture related rpg style character description, perhaps stored in a meta text file like the ones in simple_skins ... or maybe that would just be boring

Interesting idea but I guess it's only interesting for servers emphasizing role play.

**Comment by ghost** [comment]
I updated the file : changed the comment, defined h locally and used 'info' level for log messages (so it is displayed only if minestest is started with --info option )

h.s is the first number of the texture : player_hash_skin textures names starts a 0, but simple_skins ones start at 1
h.n is the count of textures, it starts at 8 (the number of textures included with player_hash_skin) 
and is redefined when counting textures, then "h.n = h.n - h.s" is called to remove 1 from count if it didn't start from 0. 
This might be a bit of paranoia but I feel better when variables are defined before going through a bunch of operations.
I guess it could be removed though... should I remove it?

**Comment by Hiradur** [comment]
> This might be a bit of paranoia but I feel better when variables are defined before going through a bunch of operations.
> I guess it could be removed though... should I remove it?

No, but change it to a value which can not be reached in normal operation such as -1 so it's clear that it's a safeguard initialization, this also helps with debugging. The 8 is misleading and causes confusion.

**Comment by ghost** [comment]
>  change it to a value which can not be reached in normal operation such as -1 so it's clear that it's a safeguard initialization

Done

**Comment by Hiradur** [comment]
Thanks.

**Pull request merged**